### DISCUSSION TITLE
Using Composer in root to install Timber in theme folder?

### DISCUSSION THREAD

**Discussion by KingWebsites** [timber/timber#2862] [Q&A] [answered]
Hi, I'm trying to create a Composer installer in the WP root folder that will install all my plugins and add Timber to my theme folder (WP and DB already created by Laragon).

The plugins seem to install to the right place but I'm not having any luck with Timber which is installing in /root/vendor/. I've tried a few variations and this isn't working.

```
"extra": {
  "installer-paths": {
    "wp-content/plugins/{$name}/": [
      "type:wordpress-plugin"
    ],
    "wp-content/themes/starter-theme/vendor/{$name}": [
      "timber/timber"
    ]
  }
},
```

Does the Timber package have a "type" I could use instead to add it to the theme's vendor directory? Let me know what you think is the best way to install Timber into my theme directory other than running composer inside the theme folder.

**Comment by nlemoine** [comment]
Hello @KingWebsites,

What's your purpose in putting Timber inside `wp-content/themes/starter-theme/vendor/`? Even located here, you will still depend on the composer autoloader that's in `{root}/vendor`.

**Reply by KingWebsites** [reply]
Hi, I'm not using Docker or Bedrock or anything fancy. As per the instructions:

https://timber.github.io/docs/v2/installation/installation/#install-timber-into-an-existing-project

> Some developers prefer to have Timber installed as a theme dependency, so they would run this command from the theme root (ex: /wp-content/themes/my-theme/).

But thinking about this more, as I only have my theme folder in the repo I think I'll place my composer.json in the theme folder and work from there.

**Reply by KingWebsites** [reply, marked as answer]
So my main quest was to migrate from the Timber plugin to having Timber in my theme folder. I'm a front-end dev with only a small understanding of Composer.

When I create a new site on dev and need to move to production I will be probably be using the All-In-One Migrator plugin which will only duplicate content within /wp-content/. I'll be FTP-ing updates. I will not be running any command line instructions on the server. This may be a similar method used by lots of people moving away from the Timber plugin who are not using Docker, Bedrock, etc..

So I've managed to get a composer.json inside my theme folder which installs Timber and adds plugins. The only thing I'm not sure about is that the /vendor/ folder in my theme folder also includes /composer/ and /symfony/ folders, although the whole directory is just 1.28MB so I'm not worried about it too much.

For anybody else migrating away from the plugin this is my composer.json file inside my theme folder. Please feel free to comment on any suggested improvemnets.

```
{
	"name": "my-company/starter-theme",
	"type": "project",
	"description": "Install Plugins and Timber",
	"repositories": [
		{
			"type": "composer",
			"url": "https://wpackagist.org",
			"only": [
				"wpackagist-plugin/*"
			]
		}
	],
	"require": {
		"timber/timber": "^2.0",
		"wpackagist-plugin/all-in-one-wp-migration": "^7.0",
		"wpackagist-plugin/wordpress-seo": "^21.0"
	},
	"prefer-stable": true,
	"extra": {
		"installer-paths": {
			"../../plugins/{$name}/": [
				"type:wordpress-plugin"
			]
		}
	},
	"config": {
		"allow-plugins": {
			"composer/installers": true
		}
	}
}
```
